### PR TITLE
[#5853] improvement(CLI): Make the entity and arguments case-insensitive

### PR DESCRIPTION
**Pull request by Abyss-lord**
### What changes were proposed in this pull request?

Make the entity and arguments case-insensitive.

### Why are the changes needed?

Fix: #5853 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

```bash
bin/gcli.sh metalake List
# output: correct output

bin/gcli.sh Metalake list
# output: correct output

bin/gcli.sh mEtalake List
# output: correct output
```

### CODE REVIEW

**Comment by Abyss-lord** [comment]
hi @justinmclean @xunliu , Could you please take a look when you have time?

File: clients/cli/src/main/java/org/apache/gravitino/cli/Main.java [hunk 1]
```diff
@@ -72,7 +73,7 @@ protected static String resolveCommand(CommandLine line) {
     if (args.length == 2) {
       String action = args[1];
```
Comment (justinmclean) [inline clients/cli/src/main/java/org/apache/gravitino/cli/Main.java:74]: Why not just do `toLowerCase(Locale.ROOT);` here? Then there is no need to modify `isValidCommand` and you can just `return action`.

Comment (justinmclean) [reply clients/cli/src/main/java/org/apache/gravitino/cli/Main.java:74]: I think you also want to use `Locale.ENGLISH` as the commands are in English, using Locale.ROOT may give unexpected results.

Comment (Abyss-lord) [reply clients/cli/src/main/java/org/apache/gravitino/cli/Main.java]: fix

**Review by justinmclean — APPROVED** [review]
LGTM thanks!

**Pull request merged**